core/src/avm2.rs

```diff
@@ -119,6 +119,10 @@ pub struct Avm2<'gc> {
     /// Scopes currently present of the scope stack.
     scope_stack: Vec<Scope<'gc>>,
 
+    /// Default XML namespace stack for E4X operations.
+    /// The last element is the current default namespace.
+    default_xml_namespace_stack: Vec<AvmString<'gc>>,
+
```

Comment:
Why use a stack here? There's no `PopDxns` opcode, so a single `default_xml_namespace: Option<AvmString<'gc>>` on `Activation` would be enough.